### PR TITLE
Generate versions HTML directly

### PR DESCRIPTION
**Pull request by Alexendoo**
Generates the page at https://rust-lang.github.io/rust-clippy/ directly rather than creating a JSON file to load

Also fixes https://github.com/rust-lang/rust-clippy/issues/13413

r? @flip1995

changelog: none

### CODE REVIEW

**Comment by flip1995** [comment]
Forgot to remove my name from the vacation list: https://github.com/rust-lang/rust-clippy/pull/13415

r? @flip1995

File: .github/deploy.sh [hunk 1]
```diff
@@ -25,16 +25,18 @@ if [[ $BETA = "true" ]]; then
...
 # Now let's go have some fun with the cloned repo
 cd out
 git config user.name "GHA CI"
 git config user.email "gha@ci.invalid"
 
+git status
+
+# Remove old versions.json, it's no longer used
+git rm --ignore-unmatch versions.json
+
```
Comment (flip1995) [inline .github/deploy.sh:39]: This seems unnecessary. This is probably a one-time commit once this PR is merged. No need to include this in the script.

File: util/versions.py [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
         return sys.maxsize
-    if v == "beta":
+    if v == "stable":
         return sys.maxsize - 1
+    if v == "beta":
+        return sys.maxsize - 2
     if v == "pre-1.29.0":
         return -1
+    if not v.startswith("rust-"):
+        return None
```
Comment (flip1995) [inline util/versions.py:19]: This removes the old versions from the index, right? I think this is a good step. And with that, I will probably follow up with removing all of them from the `gh-pages` branch. But will wait a few weeks to see if someone complains.

**Review by flip1995 — COMMENTED** [review]
LGTM overall. 2 questions remaining.

 I also want to give @xFrednet a chance to sanity check this.

File: util/versions.py (hunk 2 above)
Comment (flip1995) [reply util/versions.py:19]: Oh right. I think I deleted those `pre-1.29.0` dirs a while ago already.

**Review by flip1995 — APPROVED** [review]
LGTM. Giving xFrednet a bit more time to take a look.

**Review by xFrednet — APPROVED** [review]

**Comment by xFrednet** [comment]
Looks good to me as well, nice update :D

---

Roses are red,
Violets are blue,
@flip1995 approved,
Now bors, you do you

The lint page,
now static in nature,
loads quick,
and safer

**Comment by xFrednet** [comment]
I got too swept up in the poem xD

Now one more try:


Roses are red,
Violets are blue,
@flip1995 **approved**,
Now bors, you do you

The lint page,
now static in nature,
loads quick,
and safer

**Pull request merged**